app/Domains/Settings/ManageGiftOccasions/Services/UpdateGiftOccasionPosition.php

```diff
@@ -79,24 +75,23 @@ private function updatePosition(): void
             $this->updateDescendingPosition();
         }
 
-        DB::table('gift_occasions')
-            ->where('id', $this->giftOccasion->id)
+        $this->giftOccasion
             ->update([
                 'position' => $this->data['new_position'],
             ]);
     }
 
     private function updateAscendingPosition(): void
     {
-        DB::table('gift_occasions')
+        $this->account()->giftOccasions()
```

Comment:
@djaiss this is actually a fix, because with `DB::table` you didn't filter with `account_id`! So it updated all position...